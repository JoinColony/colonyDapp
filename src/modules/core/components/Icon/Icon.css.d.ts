export const main: string;
export const sizeExtraTiny: string;
export const sizeTiny: string;
export const sizeSmall: string;
export const sizeNormal: string;
export const sizeMedium: string;
export const sizeLarge: string;
export const sizeHuge: string;
export const themePrimary: string;
export const themeInvert: string;
