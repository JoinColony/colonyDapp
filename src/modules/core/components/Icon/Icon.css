/*
 * Base `Icon` styles
 * Everything inherits from this
 */

.main {
  display: inline-block;
}

.main svg {
  vertical-align: top;
}

.main svg use {
  pointer-events: none;
}

.main[role="button"] {
  cursor: pointer;

  &:active {
    outline: none;
  }

  &:hover {
    outline: none;
  }
}

/*
 * Sizing
 *
 * Next set of classes will define the icons's size
 */

.sizeTiny {
  height: 14px;
  width: 14px;
}

.sizeSmall {
  height: 16px;
  width: 16px;
}

.sizeNormal {
  height: 18px;
  width: 18px;
}

.sizeMedium {
  height: 28px;
  width: 28px;
}

.sizeLarge {
  height: 42px;
  width: 42px;
}

.sizeHuge {
  height: 64px;
  width: 64px;
}

.sizeImage {
  width: 660px;
}

/*
 * Themeing
 *
 * Next set of classes will define our various icon themes.
 *
 * Some of them will also have an `invert` version. This is for dark backgrounds.
 */

.themePrimary {
  composes: main;

  & svg {
    fill: var(--text);
    stroke: var(--text);
  }
}

.themeInvert {
  composes: main;

  & svg {
    fill: var(--colony-white);
    stroke: var(--colony-white);
  }
}
