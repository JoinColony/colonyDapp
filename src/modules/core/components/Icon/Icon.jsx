/* @flow */

import React from 'react';
import { injectIntl } from 'react-intl';

import type { IntlShape, MessageDescriptor, MessageValues } from 'react-intl';

import { getMainClasses } from '~utils/css';

import {
  icons as iconNames,
  multiColorIcons as multiColorIconNames,
} from './icons.json';
import styles from './Icon.css';

const displayName = 'Icon';

type Appearance = {
  theme?: 'primary' | 'invert',
  size?: 'tiny' | 'small' | 'normal' | 'medium' | 'large' | 'huge' | 'image',
};

type Props = {
  /** Appearance object */
  appearance?: Appearance,
  /** className for icon. Will override anything in appearance */
  className?: string,
  /** Name of icon sprite */
  name: string,
  /** Html title for the icon element */
  title: string | MessageDescriptor,
  /** Values for html title (react-intl interpolation) */
  titleValues?: MessageValues,
  /** SVG viewbox string */
  viewBox?: string,
  /** @ignore injected by `react-intl` */
  intl: IntlShape,
};

const getIcons = (map: Array<string>) =>
  map.reduce((prev, current) => {
    // eslint-disable-next-line global-require, import/no-dynamic-require, no-param-reassign
    prev[current] = require(`../../../../img/icons/${current}.svg`);
    return prev;
  }, {});

const icons = getIcons(iconNames);
const multiColorIcons = getIcons(multiColorIconNames);

const Icon = ({
  appearance = { size: 'normal', theme: 'primary' },
  className,
  intl: { formatMessage },
  viewBox: viewBoxOverride = '0 0 30 30',
  name,
  title,
  titleValues,
  ...props
}: Props) => {
  // Remove the theme if it's a multiColor icon
  const multiColorAppearance = multiColorIcons[name]
    ? { size: appearance.size || 'normal' }
    : null;
  const icon = icons[name] || multiColorIcons[name];
  const iconHref = typeof icon == 'object' ? `#${icon.default.id}` : icon;
  const iconTitle =
    typeof title == 'object' ? formatMessage(title, titleValues) : title;
  return (
    <i
      title={iconTitle}
      className={
        className || getMainClasses(multiColorAppearance || appearance, styles)
      }
      {...props}
    >
      <svg viewBox={viewBoxOverride}>
        <use xlinkHref={iconHref} />
      </svg>
    </i>
  );
};

Icon.displayName = displayName;

export default injectIntl(Icon);
