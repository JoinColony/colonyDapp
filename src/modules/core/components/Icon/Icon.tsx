import React, { HTMLAttributes, useCallback } from 'react';
import { MessageDescriptor, useIntl } from 'react-intl';

import { SimpleMessageValues } from '~types/index';
import { getMainClasses } from '~utils/css';

import {
  icons as iconNames,
  multiColorIcons as multiColorIconNames,
} from '../../../../img/icons.json';
import styles from './Icon.css';

const displayName = 'Icon';

type Appearance = {
  theme?: 'primary' | 'invert';
  size?:
    | 'extraTiny'
    | 'tiny'
    | 'small'
    | 'normal'
    | 'medium'
    | 'large'
    | 'huge';
};

interface Props extends Omit<HTMLAttributes<HTMLElement>, 'title'> {
  /** Appearance object */
  appearance?: Appearance;

  /** Disallow children */
  children?: never;

  /** className for icon. Will override anything in appearance */
  className?: string;

  /** Name of icon sprite */
  name: string;

  /** Html title for the icon element */
  title: string | MessageDescriptor;

  /** Values for html title (react-intl interpolation) */
  titleValues?: SimpleMessageValues;

  /** SVG viewbox string */
  viewBox?: string;
}

const getIcons = (map: string[]) =>
  map.reduce((prev, current) => {
    // eslint-disable-next-line global-require, import/no-dynamic-require, no-param-reassign
    prev[current] = require(`../../../../img/icons/${current}.svg`);
    return prev;
  }, {});

const icons = getIcons(iconNames);
const multiColorIcons = getIcons(multiColorIconNames);

const Icon = ({
  appearance = { size: 'normal', theme: 'primary' },
  className,
  viewBox,
  name,
  title,
  titleValues,
  ...props
}: Props) => {
  const { formatMessage } = useIntl();
  // Remove the theme if it's a multiColor icon
  const multiColorAppearance = multiColorIcons[name]
    ? { size: appearance.size || 'normal' }
    : null;
  const icon = icons[name] || multiColorIcons[name];
  const iconHref = typeof icon === 'object' ? `#${icon.default.id}` : icon;
  const iconTitle =
    typeof title === 'object' ? formatMessage(title, titleValues) : title;
  const getViewBox = useCallback(() => {
    switch (appearance.size) {
      case 'extraTiny':
        return '0 0 12 12';
      case 'tiny':
        return '0 0 14 14';
      case 'small':
        return '0 0 16 16';
      case 'normal':
        return '0 0 18 18';
      case 'medium':
        return '0 0 28 28';
      case 'large':
        return '0 0 42 42';
      case 'huge':
        return '0 0 64 64';
      default:
        return '0 0 18 18';
    }
  }, [appearance]);
  const viewBoxOverride = viewBox || getViewBox();
  return (
    <i
      title={iconTitle}
      className={
        className || getMainClasses(multiColorAppearance || appearance, styles)
      }
      {...props}
    >
      <svg viewBox={viewBoxOverride}>
        <use xlinkHref={iconHref} />
      </svg>
    </i>
  );
};

Icon.displayName = displayName;

export default Icon;
