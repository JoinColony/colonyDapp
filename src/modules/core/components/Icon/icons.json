{
  "icons": [
    "archive",
    "caret-right",
    "caret-down-small",
    "at-sign-circle",
    "attachment",
    "bell",
    "checklist",
    "circle-archive",
    "circle-arrow-back",
    "circle-back",
    "circle-check",
    "circle-close",
    "circle-eye",
    "circle-person",
    "circle-plus",
    "close",
    "envelope",
    "file",
    "forbidden-signal",
    "four-square",
    "graph-line",
    "home",
    "line-arrow-circle",
    "pencil",
    "person",
    "phrase",
    "return-arrow",
    "search",
    "settings",
    "sign-off",
    "square-plus",
    "tag",
    "three-dots-row",
    "wallet"
  ],
  "multiColorIcons": [
    "metamask",
    "hugging"
  ]
}
