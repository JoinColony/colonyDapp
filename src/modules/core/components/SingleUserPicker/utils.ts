import { AnyUser } from '~data/index';

/*
Extracts the required values to be used in the SingleUserPicker
on selection
*/
export const filterUserSelection = (data: AnyUser[], filterValue: string) => {
  if (!filterValue) {
    return data;
  }

  const filteredUsers = data.filter((user) => {
    const username = user.profile?.username || '';
    const walletAddress = user.profile?.walletAddress || '';

    return (
      username.toLowerCase().includes(filterValue.toLowerCase()) ||
      walletAddress.toLowerCase().includes(filterValue.toLowerCase()) ||
      user.id.toLowerCase().includes(filterValue.toLowerCase())
    );
  });

  const customUserValue: AnyUser = {
    id: 'filterValue',
    profile: {
      walletAddress: filterValue,
      displayName: filterValue,
    },
  };

  return [customUserValue].concat(filteredUsers);
};
