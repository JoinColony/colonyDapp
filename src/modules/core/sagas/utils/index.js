/* @flow */

import getDDB from './getDDB';
import getNetworkClient from './getNetworkClient';

export * from './transactions';
export { getDDB, getNetworkClient };
