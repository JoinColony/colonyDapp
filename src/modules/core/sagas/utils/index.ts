export { default as getColonyManager } from './getColonyManager';
export { default as getGasPrices } from './getGasPrices';
export { getMethod, getTransactionMethod } from './getMethod';
export { default as getWalletCategory } from './getWalletCategory';
