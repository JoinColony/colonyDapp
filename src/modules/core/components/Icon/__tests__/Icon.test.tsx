/* eslint-env jest */

import React from 'react';
import toJson from 'enzyme-to-json';

import { mountWithIntl } from '../../../../../__tests__/utils';

import Icon from '../Icon';
import { icons as iconNames } from '../icons.json';

describe('Icon component', () => {
  const mockFileName = iconNames[0];
  const mockTitle = 'The Greatest Icon Ever';
  test('Renders initial component', () => {
    const wrapper = mountWithIntl(
      <Icon name={mockFileName} title={mockTitle} />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('Works with MessageDescriptors', () => {
    const wrapper = mountWithIntl(
      <Icon
        name={mockFileName}
        title={{ id: 'iconTitle', defaultMessage: mockTitle }}
      />,
    );
    expect(toJson(wrapper)).toMatchSnapshot();
  });
  test('Shows the title value', () => {
    const wrapper = mountWithIntl(
      <Icon name={mockFileName} title={mockTitle} />,
    );
    expect(wrapper.html()).toContain(mockTitle);
  });
});
