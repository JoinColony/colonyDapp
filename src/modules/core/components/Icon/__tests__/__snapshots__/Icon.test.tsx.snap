// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component Renders initial component 1`] = `
<Icon
  name="active-task"
  title="The Greatest Icon Ever"
>
  <i
    className=""
    title="The Greatest Icon Ever"
  >
    <svg
      viewBox="0 0 30 30"
    >
      <use
        xlinkHref="#undefined"
      />
    </svg>
  </i>
</Icon>
`;

exports[`Icon component Works with MessageDescriptors 1`] = `
<Icon
  name="active-task"
  title={
    Object {
      "defaultMessage": "The Greatest Icon Ever",
      "id": "iconTitle",
    }
  }
>
  <i
    className=""
    title="The Greatest Icon Ever"
  >
    <svg
      viewBox="0 0 30 30"
    >
      <use
        xlinkHref="#undefined"
      />
    </svg>
  </i>
</Icon>
`;
